pkg/provider-local/controller/service/add.go

```diff
@@ -103,3 +106,37 @@ func matchExpressionsIstioIngressGateway(zone *string) []metav1.LabelSelectorReq
 		},
 	}
 }
+
+type nodeZoneInitializer struct {
+	multiZone *bool
+}
+
+func (nzi *nodeZoneInitializer) HasNodesInMultipleZones(ctx context.Context, c client.Client) (bool, error) {
```

Comment:
Does a unit test make sense for this?

Reply:
Sure. I will add a test.

Reply:
Added tests in https://github.com/gardener/gardener/pull/8998/commits/9577c6d6836f7914a98c01c8d70e8cb3fee96d22.